components/script/script_thread.rs

```diff
@@ -3669,6 +3669,14 @@ impl ScriptThread {
                 };
                 document.dispatch_composition_event(composition_event);
             },
+
+            GamepadEvent(gamepad_event) => {
+                let document = match self.documents.borrow().find_document(pipeline_id) {
+                    Some(document) => document,
```

Comment:
maybe you can call dispatch_gamepad_event here and drop `let document` ?

```suggestion
                    Some(document) => document.dispatch_gamepad_event(gamepad_event),
```